Tests/Rules/AvoidMultipleTypesParameter.tests.ps1

```diff
@@ -0,0 +1,25 @@
+BeforeAll {
+    $violationMessage = 'Parameter ''\$s1'' has more than one type specifier.'
+    $violationName = "PSAvoidMultipleTypesParameter"
+    $violations = Invoke-ScriptAnalyzer $PSScriptRoot\AvoidMultipleTypesParameter.ps1 | Where-Object {$_.RuleName -eq $violationName}
+    $noViolations = Invoke-ScriptAnalyzer $PSScriptRoot\AvoidMultipleTypesParameterNoViolations.ps1 | Where-Object {$_.RuleName -eq $violationName}
```

Comment:
Rather than using files and running the analyzer in `BeforeAll`, I would:

- Use inline scripts and use the `-ScriptDefinition` parameter
- Use the `-TestCases` parameter on `It` to parameterise the diagnostics you expect